target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8064-c2600.dts

```diff
@@ -124,7 +139,7 @@
 				cs-gpios = <&qcom_pinmux 20 GPIO_ACTIVE_HIGH>;
 
 				flash: m25p80@0 {
-					compatible = "s25fl256s1";
+					compatible = "s25fl256s1", "mx25u25635f", "jedec,spi-nor";
```

Comment:
only ``compatible = "jedec,spi-nor";`` here. Everything else is either unsupported or only available due to backward compatibility.

Check the comments of the source file for details: http://lxr.free-electrons.com/source/drivers/mtd/devices/m25p80.c?v=4.4#L246